ste/downloader-azureFiles_windows.go

```diff
@@ -4,13 +4,69 @@ package ste
 
 import (
 	"fmt"
+	"syscall"
 
 	"golang.org/x/sys/windows"
 )
 
-// This file implements the windows-triggered sddlAwareDownloader interface.
+// This file implements the windows-triggered smbPropertyAwareDownloader interface.
 
-func (bd *azureFilesDownloader) PutSDDL(sip ISDDLBearingSourceInfoProvider, txInfo TransferInfo) error {
+func (bd *azureFilesDownloader) PutFileTimes(sip ISMBPropertyBearingSourceInfoProvider, txInfo TransferInfo) error {
+	attribs, err := sip.GetFileSMBAttributes()
+
+	if err != nil {
+		return err
+	}
+
+	destPtr, err := syscall.UTF16PtrFromString(txInfo.Destination)
+
+	if err != nil {
+		return err
+	}
+
+	// This is a safe conversion.
+	err = windows.SetFileAttributes(destPtr, uint32(attribs))
+
+	if err != nil {
+		return err
+	}
+
+	// =========== set file times ===========
+
+	smbCreation, err := sip.GetFileSMBCreationTime()
+
+	if err != nil {
+		return err
+	}
+
+	// reviewers: We already persist last modified time (I think?) to some extent.
```

Comment:
We have preserve-last-modified-time.  I'd suggest you make it an error to set both that AND preserve-smb-attributes. (I.e. a normal "cook time" error).